File: masiwasi/app/src/main/java/models/Usuario.java
package models;

import java.io.Serializable;
import java.util.List;

public class Usuario implements Serializable {

    private String id;
    private String nombre;
    private String email;
    private String direccion;
    private String telefono;
    private String location;
    private String imageUrl;
    private List<Mascota> mascotas;

    public Usuario() {
        // Constructor vacío requerido por Firestore
    }

    public Usuario(String id, String nombre, String email, String direccion,
                   String telefono, String location, String imageUrl,
                   List<Mascota> mascotas) {

        this.id = id;
        this.nombre = nombre;
        this.email = email;
        this.direccion = direccion;
        this.telefono = telefono;
        this.location = location;
        this.imageUrl = imageUrl;
        this.mascotas = mascotas;
    }

    // Getters y setters

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getNombre() { return nombre; }
    public void setNombre(String nombre) { this.nombre = nombre; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getDireccion() { return direccion; }
    public void setDireccion(String direccion) { this.direccion = direccion; }

    public String getTelefono() { return telefono; }
    public void setTelefono(String telefono) { this.telefono = telefono; }

    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }

    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }

    public List<Mascota> getMascotas() { return mascotas; }
    public void setMascotas(List<Mascota> mascotas) { this.mascotas = mascotas; }
}

File: movil2/mozita/app/src/main/java/models/Usuario.kt
package models

import com.google.firebase.firestore.PropertyName
import java.io.Serializable

class Usuario : Serializable {

    private var id: String? = null
    private var nombre: String? = null
    @get:PropertyName("correo")
    @set:PropertyName("correo")
    var email: String? = null
    private var direccion: String? = null
    private var telefono: String? = null
    private var location: String? = null
    private var imageUrl: String? = null
    private var mascotas: List<Mascota>? = null

    // Constructor vacío requerido por Firestore
    constructor()

    constructor(
        id: String?,
        nombre: String?,
        email: String?,
        direccion: String?,
        telefono: String?,
        location: String?,
        imageUrl: String?,
        mascotas: List<Mascota>?
    ) {
        this.id = id
        this.nombre = nombre
        this.email = email
        this.direccion = direccion
        this.telefono = telefono
        this.location = location
        this.imageUrl = imageUrl
        this.mascotas = mascotas
    }

    // Getters y setters

    fun getId(): String? {
        return id
    }

    fun setId(id: String?) {
        this.id = id
    }

    fun getNombre(): String? {
        return nombre
    }

    fun setNombre(nombre: String?) {
        this.nombre = nombre
    }


    fun getDireccion(): String? {
        return direccion
    }

    fun setDireccion(direccion: String?) {
        this.direccion = direccion
    }

    fun getTelefono(): String? {
        return telefono
    }

    fun setTelefono(telefono: String?) {
        this.telefono = telefono
    }

    fun getLocation(): String? {
        return location
    }

    fun setLocation(location: String?) {
        this.location = location
    }

    fun getImageUrl(): String? {
        return imageUrl
    }

    fun setImageUrl(imageUrl: String?) {
        this.imageUrl = imageUrl
    }

    fun getMascotas(): List<Mascota>? {
        return mascotas
    }

    fun setMascotas(mascotas: List<Mascota>?) {
        this.mascotas = mascotas
    }
}